cli/cycpp.py

```diff
@@ -1547,8 +1563,11 @@ def _buildschema(self, cpptype, schematype=None, uitype=None, names=None):
             impl += '</element>'
         elif t == 'std::map':
             name = 'map'
-            if names[0] is not None:
-                name = names[0]
+            if isinstance(names[0], STRING_TYPES):
+                names[0] = [names[0], None]
```

Comment:
This needs an else clause added:

```python
        elif names[0] is None:
            names[0] = [name, None]
``` 
to handle cases where the map is not a top level state var (i.e. is nested) and the archetype dev hasn't provided a custom alias.